File: 6. Product of array except self/Solution.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var productExceptSelf = function(nums) {
    const ans = new Array(nums.length).fill(0)

    let prev = 1, next = 1;

    for(let i=0;i<nums.length;i++){
        ans[i] = prev;
        prev = prev*nums[i];
    }

    for(let i=nums.length-1;i>=0;i--){
        ans[i] = ans[i]*next;
        next = next*nums[i];
    }

    return ans;
};

File: 6. Product of array except self/Solution.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        ans = [0]*len(nums)

        prev ,next = 1, 1

        for i in range(len(nums)):
            ans[i] = prev
            prev = prev * nums[i]

        for i in range(len(nums)-1,-1,-1):
            ans[i] = ans[i]*next
            next = next*nums[i]
        
        return ans
